thirdeye/thirdeye-frontend/app/pods/rootcause/controller.js

```diff
@@ -50,10 +72,53 @@ export default Ember.Controller.extend({
     }
   ),
 
+  /**
+   * Configuration for the Settings component
+   */
+  settingsConfig: computed(
+    'model.{granularityOptions,filterOptions,compareModeOptions}',
+    function() {
+      const model = get(this, 'model');
+      const settingsOptions = ['granularityOptions', 'filterOptions', 'compareModeOptions'];
+
+      return getProperties(model, settingsOptions);
+    }
+  ),
+
   //
   // Public properties (computed)
   //
 
+  /**
+   * Hash containing all possible option values
+   * @type {Object}
+   */
+  options: Ember.computed('model', function() {
+    const model = this.get('model');
+    return getProperties(model, ['granularityOptions', 'filterOptions', 'maxTime', 'compareModeOptions']);
+  }),
+
+  /**
+   * Convert a  query value into option vlaue
+   * @param {String} key          - query param key
+   * @param {String|Object} value - query param value
+   */
+  queryToOption(key, value) {
```

Comment:
fyi this is not used in the code. Keeping it here in case we need to quickly implement a 2 way queryparams binding